aptos/api/aux-ts/src/amm/core/query.ts

```diff
@@ -45,20 +45,33 @@ export interface PoolInput {
 
 export async function pool(
   auxClient: AuxClient,
-  poolInput: PoolInput
+  { coinTypeX, coinTypeY }: PoolInput
 ): Promise<Pool | undefined> {
-  const type = poolType(auxClient.moduleAddress, poolInput);
-  const coinTypeLP = `${auxClient.moduleAddress}::amm::LP<${poolInput.coinTypeX}, ${poolInput.coinTypeY}>`;
-  const resource = await auxClient.getAccountResourceOptional(
+  let type = poolType(auxClient.moduleAddress, { coinTypeX, coinTypeY });
+  let resource = await auxClient.getAccountResourceOptional(
     auxClient.moduleAddress,
     type
   );
+  if (!resource) {
+    let t = coinTypeX;
```

Comment:
@lidatong I think here it will make sure the coinX is always the first type in the parameter, and coinY is always the second type.